### PR TITLE
Fix bug in PreInformed tallies calculation (causing negative counts)

### PR DESCRIPTION
**Pull request by johnwarden**
- **Add a test case to cover a bug in tallies calculation resulting in negative tallies**
- **Partly fix tallies bug, which fixes broken within-subjects test. But tallies test output is still not correct.**
- **'just build-shared-library' doesn't build node extension. Add timer.**
- **Don't write to Post table in simulations, to decouple from GlobalBrain.jl (which creates/updates Post). Use SimulationPost instead.**
- **Fix tallies calculation. Have to calculate preinformedVote per user, and not just take a snapshot. update tests.**

### CODE REVIEW

File: src/db-schema.jl [hunk 1]
```diff
@@ -557,54 +536,72 @@ function create_triggers(db::SQLite.DB)
...
+                , note_id
+                , count
+                , total
+            ) 
+            values (
+                new.tag_id
+                , new.post_id
+                , new.note_id
+                , (new.vote = 1)
+                , (new.vote != 0)
```
Comment (JohannesNakayama) [inline src/db-schema.jl:554]: What does this do / how does sqlite handle boolean operations like this?

Intuitively, my expectation would be that this evaluates to:

```
| new.vote | count | total |
|----------|-------|-------|
|    1     |   1   |   1   |
|    0     |   0   |   0   |
|   -1     |   0   |   0   |
```

Is that correct and if so, what does it mean wrt to InformedTallies?

Comment (johnwarden) [reply src/db-schema.jl:554]: Actually it will be:

| new.vote | count | total |
|----------|-------|-------|
|    1     |   1   |   1   |
|    0     |   0   |   0   |
|   -1     |   0   |   1   |

These values are added to the informed tally. That is, count is incremented if the vote is an upvote, and total is incremented if there is an upvote or a downvote (e.g. not a vote value of 0).

**Review by JohannesNakayama — APPROVED** [review]
